examples/mgpcg_advanced.py

```diff
@@ -1,41 +1,55 @@
 import numpy as np
-import time
 import taichi as ti
-
-real = ti.f32
-ti.init(default_fp=real, arch=ti.x64, kernel_profiler=True)
+import time
 
 
 @ti.data_oriented
 class MGPCG:
-    def __init__(self):
+    '''
+Grid-based MGPCG solver for the possion equation:
+
+    $\\nabla^2 x = k r$
+
+Credits by `@KLozes <https://github.com/KLozes>`_.
+
+See `examples/stable_fluid.py <https://github.com/taichi-dev/taichi/blob/master/examples/stable_fluid.py>`_ for usage example.
+
+.. note::
+
+    This solver only runs on CPU and CUDA backend since it requires the ``pointer`` SNode. Consider replace ``pointer`` by ``bitmasked`` on Metal, by ``dense`` on OpenGL.
```

Comment:
nit: Consider separating these notes to several lines to make each line contain not too many characters?